### PR TITLE
Added tags in general context, install slug dependency

### PR DESCRIPTION
**Pull request by hendri-tobing**
Added tags in general context, since tags will be used by many context like user_tag, post_tag, etc.  This PR also install a slug dependency.

### CODE REVIEW

File: lib/vutuv/generals.ex [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  """
+  @spec get_tag(integer) :: Tag.t() | nil
+  def get_tag(id), do: Repo.get(Tag, id)
+
+  @doc """
+  Creates a tag.
+  """
+  @spec create_tag(map) :: {:ok, Tag.t()} | changeset_error
+  def create_tag(attrs \\ %{}) do
+    downcase_name =
```
Comment (riverrun) [inline lib/vutuv/generals.ex]: Let's call `is_binary(attrs["name"])` just once - for example:

```
{downcase_name, slug_value} = if name = is_binary(attrs["name"]) do
  {String.downcase(name), Slugger.slugify_downcase(name)}
end
```

**Review by riverrun — CHANGES_REQUESTED** [review]
Just one change requested.

Could you also give me a little more information, or an example, about how we would use tags?

File: lib/vutuv/generals.ex (hunk 1 above)
Comment (hendri-tobing) [reply lib/vutuv/generals.ex]: This is done. Thank you.

**Comment by hendri-tobing** [comment]
> Just one change requested.
> 
> Could you also give me a little more information, or an example, about how we would use tags?

Initially, I thought this schema name is "skills". As Stefan advised on #621, it should be renamed as "tags" because it can be more general and also can be used for other things, for example (belongs_to) user_tag or post_tag. 

For users, user_tag can be users' skills like "Elixir", "Erlang", etc. People can endorse those user_tag.

The schema of user_tag can be like this
```
belongs_to :user
belongs_to :tag
has_many :endorsements
```

**Review by riverrun — APPROVED** [review]

**Pull request merged**